pkgs/applications/science/robotics/inav-configurator/default.nix

```diff
@@ -0,0 +1,54 @@
+{lib, stdenv, fetchurl, makeDesktopItem, nwjs, wrapGAppsHook, gsettings-desktop-schemas, gtk3 }:
+
+let
+  pname = "inav-configurator";
+  desktopItem = makeDesktopItem {
+    name = pname;
+    exec = pname;
+    icon = pname;
+    comment = "iNavFlight configuration tool";
+    desktopName = "iNav Configurator";
+    genericName = "Flight controller configuration tool";
+  };
+in
+stdenv.mkDerivation rec {
+  inherit pname;
+  version = "3.0.2";
+  src = fetchTarball {
+    url = "https://github.com/iNavFlight/${pname}/releases/download/${version}/INAV-Configurator_linux64_${version}.tar.gz";
+    sha256 = "17y8qv9wbyzdsvc1icx2p2fzv0ddkmh5v5nirchk8cadikk1fjgh";
+  };
+  icon = fetchurl {
+    url = "https://raw.githubusercontent.com/iNavFlight/inav-configurator/bf3fc89e6df51ecb83a386cd000eebf16859879e/images/inav_icon_128.png";
+    sha256 = "1i844dzzc5s5cr4vfpi6k2kdn8jiqq2n6c0fjqvsp4wdidwjahzw";
+  };
+
+  nativeBuildInputs = [ wrapGAppsHook ];
+
+  buildInputs = [ gsettings-desktop-schemas gtk3 ];
+
+  installPhase = ''
+    mkdir -p $out/bin \
+             $out/opt/${pname}
+
+    cp -r ${pname} inav-configurator $out/opt/${pname}/
+    install -m 444 -D $icon $out/share/icons/hicolor/128x128/apps/${pname}.png
+    cp -r ${desktopItem}/share/applications $out/share/
```

Comment:
```suggestion
```
We don't need to copy manually if you make the desktopItem a real attribute and then add copyDesktopItems to nativeBuildInputs

Reply:
I replaced the `cp` as described, but now I can't find a desktop file in the build result, is that expected?

Reply:
I can confirm this behavior, the desktop item does not appear in `$out` with the current expression. I think this might be cause by the way the phases are defined.
Add the following to the `installPhase`:

```
runHook preInstall

...

runHook postInstall
```

Maybe then the `copyDesktopItems` works?

Reply:
It works here
```
$ tree result
result
│
└── share
    ├── applications
        └── inav-configurator.desktop
```

Reply:
I just fixed it as recommended by @wucke13, now it works and is marked as resolved. I slowly wonder if force pushing "small fixes" is really a good idea ^^